tck/src/main/java/com/akkaserverless/javasdk/tck/model/action/ActionTckModelBehavior.java

```diff
@@ -36,68 +39,90 @@
 
   public ActionTckModelBehavior(ActionCreationContext creationContext) {}
 
-  public CompletionStage<Reply<Response>> processUnary(Request request) {
-    return Source.single(request).via(responses(actionContext())).runWith(singleResponse(), system);
+  public Effect<Response> processUnary(Request request) {
+    // Multiple request steps should be combined to give just one response, with subsequent steps
+    // taking precedence.
+    return response(request.getGroupsList(), actionContext());
   }
 
-  public CompletionStage<Reply<Response>> processStreamedIn(Source<Request, NotUsed> requests) {
-    return requests.via(responses(actionContext())).runWith(singleResponse(), system);
+  public Effect<Response> processStreamedIn(Source<Request, NotUsed> requests) {
+    // All request steps should be combined to produce a single response after the request stream
+    // completes.
+    // FIXME async + closing over stateful action context will be a pain for users
+    ActionContext actionContext = actionContext();
```

Comment:
that will be a concern indeed

Reply:
https://github.com/lightbend/akkaserverless-java-sdk/issues/268